pkg/util/utils.go

```diff
@@ -723,3 +726,145 @@ func SocketPath() (string, error) {
 	// Glue the socket path together
 	return filepath.Join(xdg, "podman", "podman.sock"), nil
 }
+
+// ImageScpOptions provide options for securely copying images to and from a remote host
+type ImageScpOptions struct {
+	// Remote determines if this entity is operating on a remote machine
+	Remote bool `json:"remote,omitempty"`
+	// File is the input/output file for the save and load Operation
+	File string `json:"file,omitempty"`
+	// Quiet Determines if the save and load operation will be done quietly
+	Quiet bool `json:"quiet,omitempty"`
+	// Image is the image the user is providing to save and load
+	Image string `json:"image,omitempty"`
+	// Transfer determines whether we are transferring an image locally
+	Transfer bool `json:"transfer,omitempty"`
+	// User is used in conjunction with Transfer to determine if a valid user was given to save from/load into
+	User string `json:"user,omitempty"`
+}
+
+// ParseImageSCPArgs returns the valid connection, source, and destination data based off of the information provided by the user
+// args is an array of the command arguments returned are two filled out source and destination options structs as well as a connections array and an error if applicable
+func ParseImageSCPArgs(args []string) (*ImageScpOptions, *ImageScpOptions, []string, error) {
+	source := ImageScpOptions{}
+	dest := ImageScpOptions{}
+
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		switch {
+		case strings.Contains(args[0], "@localhost"): // image transfer between users
+			source.User = strings.Split(args[0], "@")[0]
+			source.Transfer = true
+			if len(strings.Split(args[0], "::")) > 1 {
+				source.Image = strings.Split(args[0], "::")[1]
+			} else {
+				return nil, nil, nil, errors.New("No source image provided")
+			}
+		case strings.Contains(args[0], "::"):
+			source.Image = strings.Split(args[0], "::")[1] // this will get checked/set again once we validate connections
+			source.Remote = true
+			cliConnections = append(cliConnections, args[0])
+		default:
+			return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "a source or destination must be defined")
+		}
+	case 2:
+		switch {
+		case strings.Contains(args[0], "@localhost") || strings.Contains(args[1], "@localhost"):
+			source.Transfer = true
+			if len(strings.Split(args[0], "::")) > 1 && RemoteArgLength(args[0], 1) > 0 && len(strings.Split(args[1], "::")) > 1 && RemoteArgLength(args[1], 1) > 0 { // we cannot specify a new image name
+				return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "Image scp does not support image renaming. Please do not specify a destination name")
+			}
+			for i, arg := range args { // iterate over both of our arguments, the same logic applies to both
+				switch RemoteArgLength(arg, 1) {
```

Comment:
I don't understand this code, if both args are RemoteArgLength won't the second override the first?

Reply:
Since we know that there is only two args, you should process them separately and just format your structure.

Reply:
I think splitting this function so it only processes one arg will make this much more readable